### DISCUSSION TITLE
Query caching not working for `getResultStream()`?

### DISCUSSION THREAD

**Discussion by famod** [Blazebit/blaze-persistence#1397] [Q&A] [answered]
Hi,

is it to be expected that
```
builder.setCacheable(true).getResultList().stream().[...]
```
_does_ use query caching but
```
builder.setCacheable(true).getResultStream().[...]
```
doesn't?

**Comment by jwgmeligmeyling** [comment, marked as answer]
Although we completely delegate this to Hibernates query caching, I would indeed expect Hibernate to not cache result streams, as constructing a list of values in memory in order to write this to a cache would defeat the purpose of streaming.

If you want to use the Steam API without actually streaming the result, and thus allowing query caching, I recommend using getResultList().stream()

**Reply by beikov** [reply]
Yes, hibernate ignores caching when result streaming is requested as that uses a ScrollableResultSet behind the scenes

**Reply by famod** [reply]
Ok, thank you both!
